src/components/MapArea.js

```diff
@@ -40,84 +50,78 @@ const trailData = {
     "name": "Trail Planner Map",
     "desc": "Generated by AllTrails https://www.alltrails.com/"
   }
-}
+};
 
 
 export default function MapArea() {
-
-  const [ style, setStyle ] = useState("mapbox://styles/mapbox/outdoors-v11" );
-  const [ latitude, setLatitude ] = useState(45.393);
-  const [ longitude, setLongitude ]= useState(-75.511);
-  const [ zoom, setZoom ] = useState(15);
-  const mapContainer = useRef(null)
-	const map = useRef(null);
-	// this is where all of our map logic is going to live  
-	// adding the empty dependency array ensures that the map  
-	// is only created once  
-	useEffect(() => {    
-		// create the map and configure it    
-		// check out the API reference for more options    
-		map.current = new mapboxgl.Map({      
-			container: mapContainer.current,      
-			style,      
-			center: [longitude, latitude],      
-			zoom,    
-		})
-
-    map.current.on('load', function() {
-      map.current.addSource('trailData', {
-        'type': 'geojson',
-        'data': trailData
-      })
-  
+  const mapContainer = useRef(null);
+  const map = useRef(null);
+
+  const displayDataOnMap = (map, trailData) => {
+    if (!map.current.getSource("trailData")) {
+      map.current.addSource("trailData", {
+        type: "geojson",
+        data: trailData,
+      });
+    }
+
+    if (!map.current.getLayer("trailData")) {
       map.current.addLayer({
-        'id': 'trailData',
-        'type': 'line',
-        'source': 'trailData',
-        'layout': {},
-        'paint': {
-          'line-color': '#ec2222',
-          'line-width': 2,
-          'line-dasharray': [2, 1]
-        }
-      })
-    })
-
-		return () => map.current.remove() 
-
-	}, [style])
-
+        id: "trailData",
+        type: "line",
+        source: "trailData",
+        layout: {},
+        paint: {
+          "line-color": "#ec2222",
+          "line-width": 2,
+          "line-dasharray": [2, 1],
+        },
+      });
+    }
+  };
 
   useEffect(() => {
-    if (!map.current) return; // wait for map to initialize
-    map.current.on('move', () => {
-      setLongitude(map.current.getCenter().lng.toFixed(4));
-      setLatitude(map.current.getCenter().lat.toFixed(4));
-      setZoom(map.current.getZoom().toFixed(2));
+    // Render a map and configure it
+    map.current = new mapboxgl.Map({
+      container: mapContainer.current,
+      style: "mapbox://styles/mapbox/outdoors-v11",
+      ...DEFAULT_VIEWPORT,
+    });
+
+    map.current.on("style.load", () => {
+      displayDataOnMap(map, trailData);
     });
-  });
+
+    return () => map.current.remove();
+  }, []);
 
   const changeStyle = (styleURL) => {
-    setStyle(styleURL);
-  }
+    map.current.setStyle(styleURL);
+    displayDataOnMap(map, trailData);
```

Comment:
we have already estabilished a hook `style.load` that will be called everytime we change the style. So calling `displayDataOnMap` here is redundant.